src/libertem/udf/stddev.py

```diff
@@ -1,56 +1,133 @@
     """
     Given two sets of partitions, with sum of frames
-    and sum of variances, compute joint sum of frames
-    and sum of variances using one pass algorithm
+    and sum of variances, aggregate joint sum of frames
+    and sum of variances in destination partition using one pass
+    algorithm.
+
+    Erich Schubert and Michael Gertz. Numerically stable parallel computation of
+    (co-)variance. In `Proceedings of the 30th International Conference on
+    Scientific and Statistical Database Management - SSDBM 18`. ACM Press, 2018.
+    `doi:10.1145/3221269.3223036 <https://doi.org/10.1145/3221269.3223036>`_
+
+    cite:Schubert2018
 
     Parameters
     ----------
-    p0
-        Contains information about the first partition, including
-        sum of variances, sum of pixels, and number of frames used
+    dest_N : int
+        Number of frames aggregated in dest_sum and dest_varsum
+        The case :code:`dest_N == 0` is handled correctly.
+    dest_sum, dest_varsum
+        Aggregation buffers, will be updated: sum of pixels, sum of variances
+    src_N, src_sum, src_varsum
+        Information about the second partition
 
-    p1
-        Contains information about the second partition, including
-        sum of variances, sum of pixels, and number of frames used
 
     Returns
     -------
-    VariancePart
-        colletions.namedtuple object that contains information about
-        the merged partitions, including sum of variances,
-        sum of pixels, and number of frames used
+    N:
+        New number of frames aggregated in aggregation buffer
     """
-    if p0.N == 0:
-        return p1
-    N = p0.N + p1.N
+    if dest_N == 0:
+        dest_sum[:] = src_sum
+        dest_varsum[:] = src_varsum
+        return src_N
+    else:
+        N = dest_N + src_N
+
+        le = len(dest_sum)
+        for i in range(le):
+            dest_sum[i], dest_varsum[i] = _merge(
+                N,
+                dest_N, dest_sum[i], dest_varsum[i],
+                src_N, src_sum[i], src_varsum[i]
+            )
+    return N
 
-    # compute mean for each partitions
-    mean_A = (p0.sum_im / p0.N)
-    mean_B = (p1.sum_im / p1.N)
 
-    # compute mean for joint samples
-    delta = mean_B - mean_A
-    mean = mean_A + (p1.N * delta) / (p0.N + p1.N)
+@numba.njit(fastmath=True)
+def process_tile(tile, N0, sum_inout, var_inout):
+    '''
+    Compute sum and variance of :code:`tile` along navigation axis
+    and merge into aggregation buffers. Numerical "workhorse" for
+    :meth:`StdDevUDF.process_tile`.
 
-    # compute sum of images for joint samples
-    sum_im_AB = p0.sum_im + p1.sum_im
-
-    # compute sum of variances for joint samples
-    delta_P = mean_B - mean
-    var_AB = p0.var + p1.var + (p1.N * delta * delta_P)
+    Parameters
+    ----------
+    tile : numpy.ndarray
+        Tile with flattened signal dimension
+    N0 : int
+        Number of frames already aggegated in partition buffer
```

Comment:
"partition buffer" here means the `*_inout` parameters, right? Maybe should reference them directly.